src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const StateCopy = { ...state };
+  const res = [];
+
+  for (const key of actions) {
+    if (key.type === 'addProperties') {
+      for (const i in key.extraData) {
+        StateCopy[i] = key.extraData[i];
+      }
+      res.push({ ...StateCopy });
+    }
+
+    if (key.type === 'removeProperties') {
+      for (const i of key.keysToRemove) {
```

Comment:
Using `for...of` here is correct if `key.keysToRemove` is an array. Ensure that `keysToRemove` is indeed an array to avoid unexpected behavior.